### ISSUE TITLE
ERROR: Missing classes detected while running R8.

### ISSUE THREAD

**Issue by dharambudh1** [PhonePe/phonepe-pg-sdk-flutter#20]
I am facing an error after executing `flutter build apk --release` in the terminal.

![Pasted image](https://github.com/user-attachments/assets/962df233-5603-46a8-83fb-9902d51c8c50)

Download logs:  [error.txt](https://github.com/user-attachments/files/16371362/error.txt)

Here are my PhonePe SDK version (from: https://pub.dev/packages/phonepe_payment_sdk):
```
phonepe_payment_sdk: ^2.0.2
```

Here are my Flutter Doctor Summary:
```
[✓] Flutter (Channel stable, 3.19.3, on Ubuntu 24.04 LTS 6.8.0-38-generic, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Android Studio (version 2024.1)
```

Here are my SDK & Gradle configurations. (It complies with the latest Google Play Store policies).

build.gradle:
```
minSdkVersion 24
targetSdk 35
compileSdk 35
```

settings.gradle:
```
id "com.android.application" version "8.5.0" apply false
id 'com.google.gms.google-services' version '4.4.1' apply false
id "org.jetbrains.kotlin.android" version "1.9.23" apply false
```

gradle-wrapper.properties:
```
distributionUrl=https\://services.gradle.org/distributions/gradle-8.7-all.zip
```

**Comment by dharambudh1** [comment]
I have attempted to resolve the issue by creating a `proguard-rules.pro` file with the rule `-keep class com.phonepe.phonepe_payment_sdk.** { *; }`. I then referenced this ProGuard file in the `build.gradle` file, both of which are located at `/home/dharam/.pub-cache/hosted/pub.dev/phonepe_payment_sdk-2.0.2/android`, using the following configuration:

proguard-rules.pro:
```
-keep class com.phonepe.phonepe_payment_sdk.** { *; }
```

build.gradle:
```
proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
```

Could you please confirm if this solution is appropriate? Additionally, I would like to inquire if there are any opportunities to contribute to this project.

![Pasted image (3)](https://github.com/user-attachments/assets/906c2aeb-c9e7-42e8-97ca-9707027d3097)

![Pasted image (2)](https://github.com/user-attachments/assets/22e8f512-38e0-46fb-aa9f-5f1c8c0b35c6)

**Comment by dharambudh1** [comment]
I have attempted to resolve the issue by creating a `proguard-rules.pro` file with the rule `-keep class com.phonepe.phonepe_payment_sdk.** { *; }`. I then referenced this ProGuard file in the `build.gradle` file, both of which are located at `/home/dharam/.pub-cache/hosted/pub.dev/phonepe_payment_sdk-2.0.2/android`, using the following configuration:

proguard-rules.pro:
```
-keep class com.phonepe.phonepe_payment_sdk.** { *; }
```
![Pasted image (3)](https://github.com/user-attachments/assets/3a7c7ec0-207d-4466-b1d5-156187f48c53)

build.gradle:
```
proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
```
![Pasted image (2)](https://github.com/user-attachments/assets/087a5cbb-588e-4d1f-8ae3-e501803b2a7d)

Could you please confirm if this solution is appropriate? Additionally, I would like to inquire if there are any opportunities to contribute to this project.

**Comment by dharambudh1** [comment]
@rajanarora27, @NitinVermaPpe, @tejab41097, we can consider this issue resolved as it has been addressed in `phonepe_payment_sdk` version 2.0.3. You can review the details of the update [here](https://pub.dev/packages/phonepe_payment_sdk/versions/2.0.3).

Thank you for your attention to this matter.
